src/elf/note.rs

```diff
@@ -205,7 +205,7 @@ if_alloc! {
             };
             debug!("{:?} - {:#x}", header, *offset);
             // -1 because includes \0 terminator
-            let name = bytes.gread_with::<&'a str>(offset, ctx::StrCtx::Length(header.n_namesz - 1))?;
+            let name = bytes.gread_with::<&'a str>(offset, ctx::StrCtx::Length(header.n_namesz.saturating_sub(1)))?;
             *offset += 1;
```

Comment:
Come to think of it, this is probably wrong when n_names was 0 in the first place?

Reply:
maybe? i don't know haha; maybe the desc starts after a null-delimited empty string